addons/missile_manpad/functions/fnc_eachFrame.sqf

```diff
@@ -0,0 +1,128 @@
+        * linearConversion [100, 3000, getMass _target, 0.25, 1, true];
+
+    if (EGVAR(missileguidance,debug_drawGuidanceInfo)) then {
+        private _color = [[0,1,0,1], [0,0,1,1]] select isNull (ACE_player getVariable [QEGVAR(missileguidance,target), objNull]);
+        drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\select_target_ca.paa", _color,
+            ASLToAGL _aimASL, 1.5, 1.5, 45, format ["a%1 - v%2 - d%3 = %4", _angle, _visibility, _dist, _ret], 0.5, 0.025, "TahomaB"];
+    };
+    _ret
+};
+
+
+if (!GVAR(isLockKeyDown) || {(currentMagazine _unit) == ""}) exitWith {
+    "" call _fnc_playSound;
+    _unit setVariable [QEGVAR(missileguidance,target), nil];
+    _args set [4, objNull]; // _lockCanidate = objNull
+};
+
+private _source = _unit;
+private _seekerASL = eyePos _source;
+private _seekerDir = _source weaponDirection currentWeapon _source;
+
+private _lockFeedback = 0;
+if (isNull _lockCanidate) then {
+    // find any target within seeker range
+    private _potentialTargets = _source nearEntities ["Air", _seekerMaxRange];
+    private _bestValue = 0;
+    private _bestTarget = objNull;
+    {
+        private _target = _x;
+        if (_target isKindOf "ParachuteBase") then { continue };
+        private _strength = [_x, _lockAngle, _seekerMaxRange] call _fnc_searchTarget;
+        if (_strength > _bestValue) then {
+            _bestTarget = _x;
+            _bestValue = _strength;
+        };
+    } forEach _potentialTargets;
+    if (!isNull _bestTarget) then {
+        TRACE_1("new target",_bestTarget);
+        _args set [4, _bestTarget]; // _lockCanidate = _bestTarget;
+        _args set [5, CBA_missionTime]; // _lockStartTime = CBA_missionTime;
+        _args set [6, false]; // _haveLock = false;
+    };
+} else {
+    private _angle = [_lockAngle, _uncageAngle] select CBA_events_control;
+    private _strength = [_lockCanidate, _angle, _seekerMaxRange] call _fnc_searchTarget;
+    if (_strength == 0) exitWith {
+        _haveLock = false;
+        _args set [4, objNull]; // _lockCanidate = objNull;
+    };
+
+    if (_haveLock) then {
+        _lockFeedback = 1;
+    } else {
+        private _lockTime = CBA_missionTime - _lockStartTime;
+        private _lockNeeded = linearConversion [1, 0, _strength, _lockingTimeMin, _lockingTimeMax, true];
+        _lockFeedback = linearConversion [0, _lockNeeded, _lockTime, 0, 1, true];
+        _haveLock = _lockFeedback >= 1;
+        if (_haveLock) then {
+            TRACE_1("new lock",_strength);
+            _args set [6, true];
+        };
+    };
+
+    // hintSilent format ["lf %1\nstr %2", _lockFeedback, _strength];
+};
+
+switch (true) do {
+    case (_lockFeedback == 0): { "" call _fnc_playSound };
+    case (_lockFeedback < 1): { _lockingSound call _fnc_playSound };
+    case (_lockFeedback >= 1): { _lockedSound call _fnc_playSound };
+};
+
+
+if ((isNull _lockCanidate) || {!_haveLock}) then {
```

Comment:
```suggestion
if ((isNull _lockCandidate) || !_haveLock) then {
```
`_lockCandidate` is misspelt many times in this file.

Boolean check. This time I ain't misreading the vars.